Add Check Support for Java 17 Sealed Classes: WhitespaceAround

child of #14969 
check doc: https://checkstyle.org/checks/whitespace/whitespacearound.html#WhitespaceAround

---

```
PS D:\CS\test> javac src/Test.java                                               
PS D:\CS\test> cat config.xml                                                    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="WhitespaceAround"/>
  </module>
</module>
PS D:\CS\test> cat src/Test.java                                                 
public class Test {
    sealed interface Shape permits Circle, Rectangle{}
    final class Circle implements Shape {  }
    non-sealed class Rectangle implements Shape {  }
}
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml  src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:2:53: '{' is not followed by whitespace. [WhitespaceAround]
[ERROR] D:\CS\test\src\Test.java:2:53: '{' is not preceded with whitespace. [WhitespaceAround]
[ERROR] D:\CS\test\src\Test.java:2:54: '}' is not preceded with whitespace. [WhitespaceAround]
Audit done.
Checkstyle ends with 3 errors.
```

---

No update is needed we can just add new input

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAroundCheckTest.java
@@ -586,4 +586,21 @@ public class WhitespaceAroundCheckTest
             getNonCompilablePath("InputWhitespaceAroundLiteralWhen.java"),
             expected);
     }
+
+    @Test
+    public void testWhitespaceAroundAfterPermitsList() throws Exception {
+        final String[] expected = {
+            "25:53: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "{"),
+            "25:53: " + getCheckMessage(MSG_WS_NOT_PRECEDED, "{"),
+            "25:54: " + getCheckMessage(MSG_WS_NOT_PRECEDED, "}"),
+            "26:40: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "{"),
+            "26:40: " + getCheckMessage(MSG_WS_NOT_PRECEDED, "{"),
+            "26:41: " + getCheckMessage(MSG_WS_NOT_PRECEDED, "}"),
+            "27:48: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "{"),
+            "27:48: " + getCheckMessage(MSG_WS_NOT_PRECEDED, "{"),
+            "27:49: " + getCheckMessage(MSG_WS_NOT_PRECEDED, "}"),
+        };
+        verifyWithInlineConfigParser(
+                getNonCompilablePath("InputWhitespaceAroundAfterPermitsList.java"), expected);
+    }
 }

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespacearound/InputWhitespaceAroundAfterPermitsList.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespacearound/InputWhitespaceAroundAfterPermitsList.java
@@ -1,0 +1,28 @@
+/*
+WhitespaceAround
+allowEmptyConstructors = true
+allowEmptyMethods = (default)false
+allowEmptyTypes = (default)false
+allowEmptyLoops = (default)false
+allowEmptyLambdas = (default)false
+allowEmptyCatches = (default)false
+ignoreEnhancedForColon = (default)true
+tokens = (default)ASSIGN, BAND, BAND_ASSIGN, BOR, BOR_ASSIGN, BSR, BSR_ASSIGN, BXOR, \
+         BXOR_ASSIGN, COLON, DIV, DIV_ASSIGN, DO_WHILE, EQUAL, GE, GT, LAMBDA, LAND, \
+         LCURLY, LE, LITERAL_CATCH, LITERAL_DO, LITERAL_ELSE, LITERAL_FINALLY, \
+         LITERAL_FOR, LITERAL_IF, LITERAL_RETURN, LITERAL_SWITCH, LITERAL_SYNCHRONIZED, \
+         LITERAL_TRY, LITERAL_WHILE, LOR, LT, MINUS, MINUS_ASSIGN, MOD, MOD_ASSIGN, \
+         NOT_EQUAL, PLUS, PLUS_ASSIGN, QUESTION, RCURLY, SL, SLIST, SL_ASSIGN, SR, \
+         SR_ASSIGN, STAR, STAR_ASSIGN, LITERAL_ASSERT, TYPE_EXTENSION_AND
+
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.whitespace.whitespacearound;
+
+public class InputWhitespaceAroundAfterPermitsList {
+    sealed interface Shape permits Circle, Rectangle{} // 3 violations
+    final class Circle implements Shape{} // 3 violations
+    non-sealed class Rectangle implements Shape{} // 3 violations
+}

